test/signal_level_unittest.py

```diff
@@ -0,0 +1,159 @@
+import os
+import unittest
+
+from freezegun import freeze_time
+
+from oadr2.poll import OpenADR2
+from test.util import read_xml
+
+
+class SignalLevelTest(unittest.TestCase):
+    def setUp(self):
+        self.start_thread = False
+
+        # Some configureation variables, by default, this is for the a handler
+        config = {'vtn_ids': 'vtn_1,vtn_2,vtn_3,TH_VTN',
+                  'ven_id': 'ven_py'}
+
+        self.adr_client = OpenADR2(
+            event_config=config,
+            vtn_base_uri="",
+            start_thread=self.start_thread
+        )
+
+        # Read input data
+        # All events are scheduled on 2020-03-18
+        xml_dir = os.path.join(os.path.dirname(__file__), 'xml_files/signal_level_files')
+
+        # Status = active
+        # For period 08:00 - 13:00 value = 1
+        self.event_active_single_interval = read_xml(os.path.join(xml_dir, "event_active_single_interval.xml"))
+
+        # Status = active
+        # For period 10:00 - 14:00 value = 3
+        # For period 14:00 - 18:00 value = 2
+        self.event_active_multiple_intervals = read_xml(os.path.join(xml_dir, "event_active_multiple_intervals.xml"))
+
+        # Status = cancelled
+        # For period 08:00 - 18:00 value = 3
+        self.event_cancelled = read_xml(os.path.join(xml_dir, "event_cancelled.xml"))
+
+        # Status = near
+        # For period 20:00 - 22:00 value = 2
+        self.event_pending = read_xml(os.path.join(xml_dir, "event_pending.xml"))
+
+        # Make things a little nicer for us to see
+        print('')
+        print((40 * '='))
+
+    def tearDown(self):
+        self.adr_client.event_handler.update_all_events({}, '')  # Clear out the database
+
+    def test_active_event_with_single_interval(self):
+        print('in test_active_event_with_single_interval()')
+        events = [self.event_active_single_interval]
+
+        with freeze_time("2020-03-18 07:00:00"):
```

Comment:
I guess that time is somehow related to time of the event. Perhaps there's a way to express it here?

Reply:
Bump.